src-docs/webpack.config.js

```diff
@@ -40,149 +40,133 @@ function employCache(loaders) {
+            filename: 'images/[contenthash]-[name].[ext]',
+          },
+          parser: {
+            dataUrlCondition: {
+              maxSize: 8 * 1024,
+            },
           },
-          'css-loader',
-          'postcss-loader',
-          'sass-loader',
-        ]),
-        exclude: /node_modules/,
-      },
-      {
-        test: /\.css$/,
-        use: employCache(['style-loader', 'css-loader']),
-        exclude: /node_modules/,
-      },
-      {
-        test: /\.(woff|woff2|ttf|eot|ico)(\?|$)/,
-        loader: 'file-loader',
-      },
-      {
-        test: /\.(png|jp(e*)g|svg|gif)$/,
-        loader: 'url-loader',
-        options: {
-          limit: 8000, // Convert images < 8kb to base64 strings
-          name: 'images/[hash]-[name].[ext]',
         },
-      },
+      ],
+    },
+
+    plugins: [
+      new HtmlWebpackPlugin({
+        template: 'index.html',
+        inject: 'body',
+        cache: true,
+        showErrors: true,
+      }),
+
+      new CircularDependencyPlugin({
+        exclude: /node_modules/,
+        failOnError: true,
+      }),
+
+      new NodePolyfillPlugin(),
+      // run TypeScript during webpack build
+      // new ForkTsCheckerWebpackPlugin({
+      //   typescript: { configFile: path.resolve(__dirname, '..', 'tsconfig.json') },
+      //   async: false, // makes errors more visible, but potentially less performant
+      // }),
     ],
-  },
-
-  plugins: [
-    new HtmlWebpackPlugin({
-      template: 'index.html',
-      inject: 'body',
-      cache: true,
-      showErrors: true,
-    }),
-
-    new CircularDependencyPlugin({
-      exclude: /node_modules/,
-      failOnError: true,
-    }),
-
-    // run TypeScript during webpack build
-    // new ForkTsCheckerWebpackPlugin({
-    //   typescript: { configFile: path.resolve(__dirname, '..', 'tsconfig.json') },
-    //   async: false, // makes errors more visible, but potentially less performant
-    // }),
-  ],
-
-  devServer: isDevelopment
-    ? {
-        contentBase: 'src-docs/build',
-        host: '0.0.0.0',
-        allowedHosts: ['*'],
-        port: getPortSync({
-          port: getPort.makeRange(8030, 8130),
+
+    devServer: isDevelopment
+      ? {
+          static: {
+            directory: path.resolve(__dirname, 'build'),
+            // prevent file watching while running on CI
+            // /app/ represents the entire docker environment
+            watch: isPuppeteer ? { ignored: '**/*' } : undefined,
+          },
           host: '0.0.0.0',
-        }),
-        disableHostCheck: true,
-        historyApiFallback: true,
-        // prevent file watching while running on CI
-        // /app/ represents the entire docker environment
-        watchOptions: isPuppeteer
-          ? {
-              ignored: '**/*',
-            }
-          : undefined,
-      }
-    : undefined,
-  node: {
-    fs: 'empty',
-  },
+          port,
+          allowedHosts: 'all',
+          historyApiFallback: true,
```

Comment:
Why has the `devServer` setting changed so much? Did the API just change significantly? And do all the previous settings have an equivalent in the new settings?

Reply:
1. the `contentBase`, ..., `watchOptions` were removed in favor of the static option (https://github.com/webpack/webpack-dev-server/blob/master/CHANGELOG.md#400-beta0-2020-11-27, https://github.com/webpack/webpack-dev-server/blob/master/CHANGELOG.md#static)
2. the `disableHostCheck` and `allowedHosts` options were removed in favor of the `firewall` option (https://github.com/webpack/webpack-dev-server/blob/master/CHANGELOG.md#400-beta0-2020-11-27)
3. rename `firewall` option to `allowedHosts` option (https://github.com/webpack/webpack-dev-server/blob/master/CHANGELOG.md#400-rc0-2021-07-19)

The other fields have not changed